alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/navigationmeshes/deaccon/Deaccon2D.kt

```diff
@@ -118,47 +133,232 @@ class Deaccon2D(
+     * connected (if no obstacle is between them). Note that active seeding and
+     * cleaning phases won't be performed.
+     */
+    fun generateEnvGraph(envStart: Point2D, envWidth: Double, envHeight: Double, envObstacles: Collection<Shape>, seedsPositions: Collection<Point2D>, side: Double, destinations: Collection<Euclidean2DPosition>, crossingSide: Double? = null): NavigationGraph<Euclidean2DPosition, Euclidean2DTransformation, ConvexPolygon, Euclidean2DCrossing> =
+        generateEnvGraph(generateNavigationMeshHelper(envStart, envWidth, envHeight, envObstacles, seedsPositions, side), destinations, envObstacles, crossingSide)
+
     /*
-     * Generally speaking, obstacles may be categorized into the following categories:
-     *
-     * - Polygons (plane figures described by a finite number of straight line segments
-     * connected to form a closed polygonal chain or polygonal circuit). Polygons can
-     * be:
-     * -- convex: these are the easiest to treat, and the only supported ones
-     * -- concave: it's fairly easy to convert a concave polygon to a mesh of convex
-     * ones, so they can be treated as well
-     *
-     * - Curves (formally, curves aren't necessarily closed. In this context, however,
-     * I am referring to closed shapes with some curved segment). The algorithm does
-     * not support any type of curves at present, so curved obstacles need to be
-     * converted to polygonal ones (e.g. using a bounding box, eventually an arbitrarily
-     * oriented minimum bounding box).
-     * In the future, however, the algorithm may be modified to support this kind of
-     * obstacles. In particular, the growing phase should be modified to support curves.
-     * Different scenarios involving both curved convex obstacles and concave ones
-     * need to be taken into account. This possible future modification is the only reason
-     * why this method does not take a Collection<ConvexPolygon> as input for obstacles.
+     * This is a basic algorithm for generating an environment's graph.
+     * It requires no degenerate edge or collinear points in the walkable areas.
      */
-    fun generateNavigationMesh(envStart: Point2D, envWidth: Double, envHeight: Double, envObstacles: Collection<Shape>): Collection<ConvexPolygon> {
+    private fun generateEnvGraph(navMesh: Pair<Collection<ExtendableConvexPolygon>, Double>, destinations: Collection<Euclidean2DPosition>, envObstacles: Collection<Shape>, crossingSide: Double? = null): NavigationGraph<Euclidean2DPosition, Euclidean2DTransformation, ConvexPolygon, Euclidean2DCrossing> {
+        val walkableAreas = navMesh.first
+        val step = crossingSide ?: navMesh.second
+        val builder = NavigationGraphBuilder<Euclidean2DPosition, Euclidean2DTransformation, ConvexPolygon, Euclidean2DCrossing>(walkableAreas.size)
+        walkableAreas.forEach { builder.addNode(it) }
+        walkableAreas.forEachIndexed { aIndex, a ->
```

Comment:
a -> area

in general, avoid short variable name, use expressive names (r, i, and so on should be avoided where reasonable)